nautobot/extras/tests/test_plugins.py

```diff
@@ -122,14 +122,24 @@ def test_git_datasource_contents(self):
         """
         registered_datasources = registry.get("datasource_contents", {}).get("extras.gitrepository", [])
 
-        self.assertIn(
-            DatasourceContent(
-                name="text files",
-                content_identifier="dummy_plugin.textfile",
-                icon="mdi-note-text",
-                callback=refresh_git_text_files,
-            ),
-            registered_datasources,
+        plugin_datasource = DatasourceContent(
+            name="text files",
+            content_identifier="dummy_plugin.textfile",
+            icon="mdi-note-text",
+            weight=1000,
+            callback=refresh_git_text_files,
+        )
+
+        for datasource in registered_datasources:
+            if datasource.name == plugin_datasource.name:
+                self.assertEqual(datasource.content_identifier, plugin_datasource.content_identifier)
+                self.assertEqual(datasource.icon, plugin_datasource.icon)
+                self.assertEqual(datasource.weight, plugin_datasource.weight)
+                self.assertEqual(datasource.callback, plugin_datasource.callback)
+                return
```

Comment:
Why `return` here? It's not clear to me from these assertions why you would want to exit the test case under this condition, especially w/ the `assertEqual` just below. Can you please add some comments for future us?